crates/core-new/src/compiler/mod.rs

```diff
@@ -0,0 +1,62 @@
+use std::collections::HashSet;
+
+use anyhow::Result;
+use hdp_primitives::{
+    datalake::task::DatalakeCompute,
+    module::Module,
+    task::{ExtendedTask, TaskEnvelope},
+};
+use hdp_provider::key::FetchKeyEnvelope;
+use module::ModuleCompilerConfig;
+
+pub mod datalake;
+pub mod module;
+
+pub struct Compiler {
+    pub(crate) datalake: datalake::DatalakeCompiler,
+    pub(crate) module: module::ModuleCompiler,
+}
+
+impl Compiler {
+    pub fn new(config: ModuleCompilerConfig) -> Self {
+        Self {
+            datalake: datalake::DatalakeCompiler::new(),
+            module: module::ModuleCompiler::new_with_config(config),
+        }
+    }
+
+    pub async fn compile(
+        &self,
+        tasks: Vec<TaskEnvelope>,
+    ) -> Result<(HashSet<FetchKeyEnvelope>, Vec<ExtendedTask>)> {
+        let (datalake_tasks, module_tasks) = self.split_tasks(tasks);
```

Comment:
could be done more elegantly with iterator approach
not important

Reply:
compiler impl detail, can resolve later PR